### PR TITLE
511 add laser distance meter

### LINKED ISSUE #511: Add Laser distance meter AR500
Add temporary support for Laser distance meter AR500. Currently supported using Galil Dig out and Analog in. 

Need modeLaser (modified modeUSAXS etc), which will move instrument to Open beam path and then move dx and dy to positions in PVs: 9idcLAX:USAXS:Laser_dx and 9idcLAX:USAXS:Laser_dy.
Switch on laser using 9idcRIO:Galil2bo1_CMD set to 1 (off is 0). 
Distance is available on : 9idcLAX:userCalc7.VAL
modify other modes to switch off laser 9if it is on).

### PR DESCRIPTION
**Pull request by jilavsky**
resolves #511

### CODE REVIEW

File: profile_bluesky/startup/instrument/devices/laser.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    uses Galil userAve (2 seconds long) to average noise
+    uses userCalc7 in LAX to convert to real distace
+    to read distace use laser.distance.get()
+    to switch on use laser.enable(1)
+    to switch off use laser.enable(0)
+    dx_in and dy_in reads defined positions from epics. PVs and on Parameters GUI. 
+    """
+    distance    = Component(EpicsSignalRO, "9idcLAX:userCalc7.VAL")
+    enable      = Component(EpicsSignal, "9idcRIO:Galil2Bo1_CMD")
+    dx_in       = Component(EpicsSignalRO, "9idcLAX:USAXS:Laser_dx")
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/laser.py:32]: Is it necessary to have `_in` suffix?  Can you get by without it?

File: profile_bluesky/startup/instrument/devices/laser.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class LaserAR500(Device):
+    """
+    Temporary support for laser distacne meter AR500
+    https://www.acuitylaser.com/product/laser-sensors/short-range-sensors/ar500-laser-position-sensor/
+    uses Galil analog in and digital out to read distacne as analog volatge and switch AR500 on/off 
+    uses Galil userAve (2 seconds long) to average noise
+    uses userCalc7 in LAX to convert to real distace
+    to read distace use laser.distance.get()
+    to switch on use laser.enable(1)
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/laser.py:26]: `laser.enable.put(1)`

File: profile_bluesky/startup/instrument/devices/laser.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    to switch off use laser.enable(0)
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/laser.py:27]: `laser.enable.put(0)`

File: profile_bluesky/startup/instrument/plans/mode_changes.py [hunk 4]
```diff
@@ -62,6 +64,25 @@ def confirm_instrument_mode(mode_name):
...
+    yield from user_data.set_state_plan(
+        "Preparing for Laser distacne meter mode"
+        )
+    yield from bps.mv(
+        ccd_shutter,        "close",
+        d_stage.x, laser.dx_in.get(),
+        d_stage.y, laser.dy_in.get(),
+        )
+    yield from bps.mv(
+        laser.enable,  1,
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/plans/mode_changes.py:81]: correct!

**Review by prjemian — CHANGES_REQUESTED** [review]
as noted

File: profile_bluesky/startup/instrument/devices/laser.py (hunk 1 above)
Comment (jilavsky) [reply profile_bluesky/startup/instrument/devices/laser.py:32]: _in is not necessary.

**Comment by jilavsky** [comment]
@prjemian comments resolved. Ready to merge?

**Review by prjemian — APPROVED** [review]
LGTM

**Pull request merged**